### DISCUSSION TITLE
fzf + fish shell not working as expected?

### DISCUSSION THREAD

**Discussion by Ncage1974** [junegunn/fzf#4129] [General]
So i'm just getting started with `fzf`. I am using the fish shell (maybe that is my issue):
when i do anything **<tab> this is what happens (ls **<tab>)):
<img width="1133" alt="image" src="https://github.com/user-attachments/assets/5dc52931-95a1-4053-8e40-d9fd831257b1" />

so it just list all the files in that directory & every directory below it. Is this because i'm using fish shell or is there something else i'm doing wrong?

Thanks....

**Comment by junegunn** [comment]
We don't provide fuzzy completion for fish, only for bash and zsh.

https://github.com/junegunn/fzf?tab=readme-ov-file#fuzzy-completion-for-bash-and-zsh

**Reply by Ncage1974** [reply]
@junegunn thanks. I appreciate the fast reply Wasn't sure how supported the fish shell was. I love fish but this might just push me over the edge to go back to zsh. There is just to many compatibility issues with fish.

**Comment by lalvarezt** [comment]
have you tried setting up ctrl-T integration for fish? This offers the same functionality (at least for me) as **\<Tab\>

https://github.com/junegunn/fzf/blob/9c1a47acf7453f9dad5905b7f23ad06e5195d51f/shell/key-bindings.fish#L134-L151

https://github.com/junegunn/fzf/blob/9c1a47acf7453f9dad5905b7f23ad06e5195d51f/shell/key-bindings.fish#L220-L223

Mentioned in the docs
```fish
# Set up fzf key bindings
fzf --fish | source
```
